### ISSUE TITLE
Invalid Hook call in dynamic routes

### ISSUE THREAD

**Issue by kush-daga** [netlify/netlify-plugin-nextjs#607]
**Describe the bug**
In my next.js website, the dynamic routes, are throwing an error.

![image](https://user-images.githubusercontent.com/40840079/130247032-1dbd6e56-eb79-4d12-970c-7125e57dbe48.png)

These work fine locally, and give no errors.
This can be related to using of `esbuild` as node_bundler.

**Expected behavior**
The dynamic routes should work, as they do locally.

**Versions**
- Next.js: ^10.2.0
- plugin (if installed at fixed version):

**If you're using the CLI to build**
 - OS: [e.g. macOS, Windows]
 - netlify/cli version:

**If you're using file-based installation**
- netlify.toml: 
![image](https://user-images.githubusercontent.com/40840079/130247692-2c45e0d8-27ce-4266-bc58-7606f0ef2717.png)

**Comment by kush-daga** [comment]
Definitely seems like a netlify plugin issue, as tested the same by deploying on vercel, and it works as expected

**Comment by lindsaylevine** [comment]
@kush-daga hey! thanks for reporting this to us! can you please provide a simple and deployable repo that reproduces your issue? it can have just a single failing dynamic page. thank you!

**Comment by kush-daga** [comment]
> @kush-daga hey! thanks for reporting this to us! can you please provide a simple and deployable repo that reproduces your issue? it can have just a single failing dynamic page. thank you!

Heyy! Thanks for your reply, I did recreate this issue, 

This is the url -> https://romantic-gates-22561a.netlify.app/dynamic/1 
GitHub Repo -> https://github.com/kush-daga/next-netlify-bug

**Comment by kush-daga** [comment]
Also I have the NETLIFY_EXPERIMENTAL_PROCESS_DYNAMIC_IMPORTS set to true in my netlify env variables

**Comment by lindsaylevine** [comment]
hey @kush-daga! switching to target: serverless from target: experimental-serverless-trace should fix your issue. let me know and we can close :)!

**Comment by lindsaylevine** [comment]
and ftr, the `NETLIFY_EXPERIMENTAL_PROCESS_DYNAMIC_IMPORTS` is no longer needed for esbuild to work with dynamic imports!

**Comment by kush-daga** [comment]
> hey @kush-daga! switching to target: serverless from target: experimental-serverless-trace should fix your issue. let me know and we can close :)!

Hey, umm I wanted to do this, however I am using sentry in my app, which requires this mode of target, as per this comment https://github.com/getsentry/sentry-javascript/issues/3621#issuecomment-850879801

**Comment by lindsaylevine** [comment]
hmm. i pinged our esbuild expert internally to see what might be up. in the meantime, is it possible to omit one of the three offending features? (aka e-s-t, esbuild, or sentry?) or are all three required for your use case?

note that we will "soon" be moving off of target: serverless to target: server. not in the very near future, but eventually.

**Comment by kush-daga** [comment]
> hmm. i pinged our esbuild expert internally to see what might be up. in the meantime, is it possible to omit one of the three offending features? (aka e-s-t, esbuild, or sentry?) or are all three required for your use case?
> 
> note that we will "soon" be moving off of target: serverless to target: server. not in the very near future, but eventually.

Hmmm, I wish i could omit one, but seems like all three are required for now, thanks for the reply though, will be waiting on an update :)

**Comment by ascorbic** [comment]
Hi @kush-daga
I managed to reproduce this issue. Removing `next` from external node modules fixes it for me. Was there a reason you needed to add it? Let me know if it works for you.

**Comment by ascorbic** [comment]
Ah, I realise the error message was suggesting it. We will fix that. To be clear: you shouldn't add `next` to external node modules, whatever the message says.
